### PR TITLE
Simon

### PR DESCRIPTION
**Pull request by simonmogli**
Please review

### CODE REVIEW

File: test/zombieHelper.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  let zombieCon;
+
+
+  beforeEach(async ()  => {
+    zombieCon = await ZombieHelper.deployed();
+  });
+  //get the level of a zombie with specific id
+  //returns -1 if it fails
+  let getZombieLevel = (id) => {
+    return ZombieHelper.deployed().then((instance)=>{
```
Comment (alexlatif) [inline test/zombieHelper.js]: we need to move this into beforeEach hook

File: test/zombieHelper.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      zombieCon = instance;
+      return zombieCon.zombies.call(id).then((result)=>{return result[0];}).catch(()=>{return ""});
+    });
+  }
+
+  //get the dna of a zombie with specific id
+  //returns -1 if it fails
+  let getZombieDna = (id) =>{
+    return ZombieHelper.deployed().then((instance)=>{
```
Comment (alexlatif) [inline test/zombieHelper.js]: because it is written twice

**Review by alexlatif — CHANGES_REQUESTED** [review]
We need to clean up the commits with rebasing

make sure to use the beforeEach hook

File: migrations/2_deploy_contracts.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
...
+const  ZombieAttack=artifacts.require("ZombieAttack");
+const SafeMath=artifacts.require("SafeMath");
 
-// module.exports = function(deployer) {
-//   deployer.deploy(ConvertLib);
-//   deployer.link(ConvertLib, MetaCoin);
-//   deployer.deploy(MetaCoin);
-// };
+module.exports = function(deployer) {
+  deployer.deploy(SafeMath);
```
Comment (scriptkitty) [inline migrations/2_deploy_contracts.js:5]: Are we sure now that this is the right approach? What about the updateContract routines?

**Review by scriptkitty — APPROVED** [review]

File: migrations/2_deploy_contracts.js (hunk 3 above)
Comment (alexlatif) [reply migrations/2_deploy_contracts.js:5]: The correct deployement script is on master. This file will be ignored in the merge. What are your thought on the update contract routine?

I was of the opinion that we have a suicide function within the solidity contract and if the contract address is changed then we call suicide on the previous address

**Pull request closed without merging**